vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -495,17 +498,30 @@ def collect_events(self):
             self.log.debug("Got %s new events from the vCenter event manager", len(new_events))
             event_config = {'collect_vcenter_alarms': True}
             for event in new_events:
+                if event.key in self.latest_processed_events:
+                    self.log.debug(
+                        "Skip already processed event (current event key: %s, latest processed events: %s).",
+                        event.key,
+                        self.latest_processed_events,
+                    )
+                    continue
+                self.log.debug("Process event (key=%s)", event.key)
+                latest_processed_events.append(event.key)
                 normalized_event = VSphereEvent(event, event_config, self.config.base_tags)
                 # Can return None if the event if filtered out
                 event_payload = normalized_event.get_datadog_payload()
                 if event_payload is not None:
                     self.event(event_payload)
+                if latest_event is None or event.createdTime > latest_event.createdTime:
+                    latest_event = event
         except Exception as e:
             # Don't get stuck on a failure to fetch an event
             # Ignore them for next pass
             self.log.warning("Unable to fetch Events %s", e)
 
-        self.latest_event_query = self.api.get_latest_event_timestamp() + dt.timedelta(seconds=1)
+        if latest_event is not None:
+            self.latest_event_query_time = latest_event.createdTime
+            self.latest_processed_events = latest_processed_events
```

Comment:
Using `+ dt.timedelta(seconds=1)` is not safe since we will miss events between latest event and +1 seconds.